subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -234,6 +237,72 @@ static int handle_device_config_update(const struct nct_evt *const evt,
 	return err;
 }
 
+/* Placeholder until cloud logging added in another PR */
+void nrf_cloud_log_control(int log_level)
+{
+	LOG_DBG("Setting nRF Cloud log level = %d", log_level);
+}
+
+static int handle_device_control_update(const struct nct_evt *const evt,
+					bool *const control_found)
+{
+	int err;
+	struct nct_cc_data msg = {
+		.opcode = NCT_CC_OPCODE_UPDATE_REQ,
+		.message_id = NCT_MSG_ID_STATE_REPORT,
+	};
+	struct nrf_cloud_evt cloud_evt = {
+		.type = NRF_CLOUD_EVT_RX_DATA_SHADOW
+	};
+	enum nrf_cloud_ctrl_status status = NRF_CLOUD_CTRL_NOT_PRESENT;
+	struct nrf_cloud_ctrl_data ctrl_data;
+
+	if ((evt == NULL) || (control_found == NULL)) {
+		return -EINVAL;
+	}
+
+	if (evt->param.cc == NULL) {
+		return -ENOENT;
+	}
+
+	err = nrf_cloud_decode_control(&evt->param.cc->data, &status, &ctrl_data);
+	if (err) {
+		return (err == -ESRCH) ? 0 : err;
+	}
+
+	*control_found = (status != NRF_CLOUD_CTRL_NOT_PRESENT);
+	if (*control_found) {
+		nrf_cloud_alert_control(ctrl_data.alerts_enabled);
+		nrf_cloud_log_control(ctrl_data.log_level);
```

Comment:
If a message comes in that specifies a state for only one of the two controllable states, the other state will reset to false -- Consider initializing the `ctrl_data` struct with their respective current values

Reply:
good catch